python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_graph/_digraph_group_chat.py

```diff
@@ -37,12 +37,42 @@ class DiGraphEdge(BaseModel):
     """
 
     target: str  # Target node name
-    condition: str | None = None  # Optional execution condition (trigger-based)
+    condition: Union[str, Callable[[BaseChatMessage], bool], None] = None
```

Comment:
Let's remove the str option from the Union. it will be a breaking change, but the GraphFlow API is experimental, so it is expected.